VideoLocker/src/main/java/org/edx/mobile/view/adapters/CourseUnitPagerAdapter.java

```diff
@@ -0,0 +1,83 @@
+package org.edx.mobile.view.adapters;
+
+
+import android.support.v4.app.Fragment;
+import android.support.v4.app.FragmentManager;
+import android.support.v4.app.FragmentStatePagerAdapter;
+
+import org.edx.mobile.model.api.EnrolledCoursesResponse;
+import org.edx.mobile.model.course.BlockType;
+import org.edx.mobile.model.course.CourseComponent;
+import org.edx.mobile.model.course.HtmlBlockModel;
+import org.edx.mobile.model.course.VideoBlockModel;
+import org.edx.mobile.view.CourseUnitDiscussionFragment;
+import org.edx.mobile.view.CourseUnitEmptyFragment;
+import org.edx.mobile.view.CourseUnitFragment;
+import org.edx.mobile.view.CourseUnitMobileNotSupportedFragment;
+import org.edx.mobile.view.CourseUnitOnlyOnYoutubeFragment;
+import org.edx.mobile.view.CourseUnitVideoFragment;
+import org.edx.mobile.view.CourseUnitWebViewFragment;
+
+import java.util.List;
+
+public class CourseUnitPagerAdapter extends FragmentStatePagerAdapter {
+    private List<CourseComponent> unitList;
+    private EnrolledCoursesResponse courseData;
+    private CourseUnitFragment.HasComponent callback;
+
+    public CourseUnitPagerAdapter(FragmentManager manager,
+                                  List<CourseComponent> unitList,
+                                  EnrolledCoursesResponse courseData,
+                                  CourseUnitFragment.HasComponent callback) {
+        super(manager);
+        this.unitList = unitList;
+        this.courseData = courseData;
+        this.callback = callback;
+    }
+
+    public CourseComponent getUnit(int pos) {
+        if (pos >= unitList.size())
+            pos = unitList.size() - 1;
+        if (pos < 0)
+            pos = 0;
+        return unitList.get(pos);
+    }
+
+    @Override
+    public Fragment getItem(int pos) {
+        CourseComponent unit = getUnit(pos);
+        CourseUnitFragment unitFragment;
+        //FIXME - for the video, let's ignore studentViewMultiDevice for now
+        if (unit instanceof VideoBlockModel &&
+                ((VideoBlockModel) unit).getData().encodedVideos.getPreferredVideoInfo() != null) {
+            unitFragment = CourseUnitVideoFragment.newInstance((VideoBlockModel) unit);
+        } else if (unit instanceof VideoBlockModel &&
+                ((VideoBlockModel) unit).getData().encodedVideos.getYoutubeVideoInfo() != null) {
+            unitFragment = CourseUnitOnlyOnYoutubeFragment.newInstance(unit);
+        } else if (unit.getType() == BlockType.DISCUSSION) {
```

Comment:
This should be changed to `unit instanceof DiscussionBlockModel`, since the parser checks the availability of the fields as well in order to create this model.